src/inference/include/ie/details/ie_blob_iterator.hpp

```diff
@@ -10,17 +10,29 @@
 
 #pragma once
 
+#if !defined(IN_OV_COMPONENT) && !defined(IE_LEGACY_HEADER_INCLUDED)
+#    define IE_LEGACY_HEADER_INCLUDED
```

Comment:
should we undef at the end? otherwise, only first header will print warning:
```c++
#include <ie_header1.hpp> // prints
#include <ie_header2.hpp> // misses
```